Implement handler to transfer LSK token (#3)
LiskHQ/lsk-faucet: README.md

# lsk-faucet

[![Report](https://goreportcard.com/badge/github.com/liskhq/lsk-faucet)](https://goreportcard.com/report/github.com/liskhq/lsk-faucet)
[![Go](https://img.shields.io/github/go-mod/go-version/liskhq/lsk-faucet)](https://go.dev/)
[![License](https://img.shields.io/github/license/LiskHQ/lsk-faucet)](https://github.com/liskhq/lsk-faucet/blob/main/LICENSE)
![GitHub repo size](https://img.shields.io/github/repo-size/liskhq/lsk-faucet)
![GitHub issues](https://img.shields.io/github/issues-raw/liskhq/lsk-faucet)
![GitHub closed issues](https://img.shields.io/github/issues-closed-raw/liskhq/lsk-faucet)

> LSK faucet is a web application to get Lisk (LSK) tokens on the Lisk Sepolia Testnet. The tokens can be used to test and troubleshoot your decentralized application or protocol before going live on the Lisk Mainnet.

Reviewer: here is the same. LSK is only used for staking/governance/claiming unless some dApps are building something using LSK token

# lsk-faucet

[![Report](https://goreportcard.com/badge/github.com/liskhq/lsk-faucet)](https://goreportcard.com/report/github.com/liskhq/lsk-faucet)
[![Go](https://img.shields.io/github/go-mod/go-version/liskhq/lsk-faucet)](https://go.dev/)
[![License](https://img.shields.io/github/license/LiskHQ/lsk-faucet)](https://github.com/liskhq/lsk-faucet/blob/main/LICENSE)
![GitHub repo size](https://img.shields.io/github/repo-size/liskhq/lsk-faucet)
![GitHub issues](https://img.shields.io/github/issues-raw/liskhq/lsk-faucet)
![GitHub closed issues](https://img.shields.io/github/issues-closed-raw/liskhq/lsk-faucet)

> LSK faucet is a web application to get sepolia Lisk (LSK) in order to test and troubleshoot your decentralized application or protocol before going live on Lisk mainnet.

Reviewer 2: ```suggestion
LSK faucet is a web application to get Lisk (LSK) tokens on the Lisk Sepolia Testnet. The tokens can be used to test and troubleshoot your decentralized application or protocol before going live on the Lisk Mainnet.
```

> This faucet is designed to provide LSK test tokens to the developers who need to test their smart contracts and interact with the blockchain.

Reviewer 2: This statement is incorrect since the transaction token on Lisk L2 (both Mainnet & Sepolia Testnet) is bridged ETH. Let's please update/remove it.
@shuse2 could you please suggest why would someone need LSK tokens on the testnet, other than for governance which in my understanding is only used by our contracts?
```suggestion
This faucet is designed to provide LSK test tokens to the developers who need to test their smart contracts and interact with the blockchain.
```

Author: I removed the statement.

Then run the faucet application without the wallet command-line flags:
```bash
./lsk-faucet -httpport 8080
```

**Optional Flags**

The following are the available command-line flags(excluding above wallet flags):

| Flag              | Description                                      | Default Value |
| ----------------- | ------------------------------------------------ | ------------- |
| -httpport         | Listener port to serve HTTP connection           | 8080          |
| -proxycount       | Count of reverse proxies in front of the server  | 0             |

> | -token-address    | Token contract address                           |               |

Reviewer 2: We should already specify the contract address here? However, I'm unsure if we should specify the L1/L2/both contracts. Please double check.
Please refer [this](https://github.com/LiskHQ/lisk-contracts/blob/main/deployment/addresses/sepolia/addresses.md) documentation.
```suggestion
| -token-address    | Token contract address                           |               |
```

| -faucet.amount    | Number of LSK to transfer per user request       | 1             |
| -faucet.minutes   | Number of minutes to wait between funding rounds | 1440          |
| -faucet.name      | Network name to display on the frontend          | sepolia       |
| -faucet.symbol    | Token symbol to display on the frontend          | LSK           |
| -hcaptcha.sitekey | hCaptcha sitekey                                 |               |
| -hcaptcha.secret  | hCaptcha secret                                  |               |

### Docker deployment
#### Build docker image
Run the following command to build docker image:
```bash
docker build -t liskhq/lsk-faucet .
```

#### Run faucet
Run the following command to start the application:

```bash

> docker run -d -p 8080:8080 -e WEB3_PROVIDER=rpc endpoint -e PRIVATE_KEY=hex private key liskhq/lsk-faucet
> ```

Reviewer 2: ````suggestion
docker run -d -p 8080:8080 -e WEB3_PROVIDER=<rpc-endpoint> -e PRIVATE_KEY=<hex-private-key> liskhq/lsk-faucet
```
**NOTE**: Please replace `<rpc-endpoint>` and `<hex-private-key>` with appropriate values.
````

or

```bash

> docker run -d -p 8080:8080 -e WEB3_PROVIDER=rpc endpoint -e KEYSTORE=keystore path -v `pwd`/keystore:/app/keystore -v `pwd`/password.txt:/app/password.txt liskhq/lsk-faucet
> ```

Reviewer 2: ````suggestion
docker run -d -p 8080:8080 -e WEB3_PROVIDER=<rpc-endpoint> -e KEYSTORE=<keystore-path> -v `pwd`/keystore:/app/keystore -v `pwd`/password.txt:/app/password.txt liskhq/lsk-faucet
```
**NOTE**: Please replace `<rpc-endpoint>` and `<keystore-path>` with appropriate values.
````
